functions/src/utils.ts

```diff
@@ -1,8 +1,34 @@
 import * as admin from 'firebase-admin'
 
+import { chunk } from 'lodash'
 import { Contract } from '../../common/contract'
 import { PrivateUser, User } from '../../common/user'
 
+export const log = (...args: unknown[]) => {
+  console.log(`[${new Date().toISOString()}]`, ...args)
+}
+
+export const logMemory = () => {
+  const used = process.memoryUsage()
+  for (const [k, v] of Object.entries(used)) {
+    log(`${k} ${Math.round((v / 1024 / 1024) * 100) / 100} MB`)
+  }
+}
+
+export const mapAsync = async <T, U>(
+  xs: T[],
+  fn: (x: T) => Promise<U>,
+  concurrency = 100
+) => {
+  const results = []
+  const chunks = chunk(xs, concurrency)
+  for (let i = 0; i < chunks.length; i++) {
+    log(`${i * concurrency}/${xs.length} processed...`)
+    results.push(...(await Promise.all(chunks[i].map(fn))))
+  }
+  return results
+}
```

Comment:
Compared to `batchedWaitAll`, obviously I wanted the logging, but I also think this signature is more ergonomic.